src/modules/graphql/resolvers/student.resolver.ts

```diff
@@ -1,46 +1,87 @@
 import { Resolver, Query, Mutation, ResolveProperty } from '@nestjs/graphql';
 import { UsersPersistenceService } from '../../persistence/users.persistence.service';
 import { ClassPersistenceService } from '../../persistence/class.persistence.service';
-import { UserRole } from '../../../models/user.db.model';
+import { UserDbModel, UserRole} from '../../../models/user.db.model';
+import {Asset, checkAndGetBasePermission, DBOperation, NO_PERMISSION, Permission} from '../../permissions/permission.interface';
+import { Get } from '../../../utils/get';
 
 @Resolver('Student')
 export class StudentResolver {
     constructor(private usersPersistence: UsersPersistenceService, private classPersistence: ClassPersistenceService) { }
 
     @Query('students')
-    async getStudents() {
+    async getStudents(_, {}, context) {
+        const permission = checkAndGetBasePermission(Get.getObject(context, 'user'), DBOperation.READ, Asset.STUDENT);
+        if (permission === Permission.OWN) {
+            // find students in teacher's class
+            const teacher: UserDbModel = await this.usersPersistence.getById(context.user.id);
+            const teacherClassId = teacher.class_id ? teacher.class_id.toString() : '';
+            const [, students] = await this.usersPersistence.getStudentsByClassId(teacherClassId);
+            // console.log(students);
+            return students;
+        }
+
         return this.usersPersistence.getUsersByFilters({role: UserRole.STUDENT});
     }
 
     @Query('student')
-    async getStudentById(_, args) {
+    async getStudentById(_, args, context) {
+        const permission = checkAndGetBasePermission(Get.getObject(context, 'user'), DBOperation.READ, Asset.STUDENT);
+        if (permission === Permission.OWN) {
+            // find student in teacher's class
+            const teacher: UserDbModel = await this.usersPersistence.getById(context.user.id);
+            const teacherClassId = teacher.class_id ? teacher.class_id.toString() : '';
+            const [, students] = await this.usersPersistence.getStudentsByClassId(teacherClassId);
+            return students.find((obj) => obj._id.toString() === args.id.toString());
+        }
```

Comment:
Here we have a lot of redundant code. getStudentById is a private case of getStudents. Can we refactor it?

Reply:
will be refactored with a new service